### PR TITLE
RHTAP-2558 - Setup dynamic e2e IntegrationTestingScenario

### PR DESCRIPTION
**Pull request by rhopp**
This PR adds new Pipeline to be used as IntegrationTestScenario. This pipeline will allow us to specify "test plan" in form of Pict model file. This new pipeline reads this file, generates test configuration matrix based on that model and then runs full E2E scenario (from cluster creation to cluster teardown) for each config.

Current logic only supports specifying different cluster versions (other parameters from the pict model are basically ignored). This functionality will be added in future in different JIRAs/PRs.

For more info see `integration-tests/README.md`

### CODE REVIEW

File: integration-tests/pipelines/e2e-main-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          - name: configs-json
+            description: Config json generated from pict file.
+        steps:
+          - name: pict-generate
+            image: quay.io/apodhrad/pict:latest
+            script: |
+              #!/usr/bin/env bash
+              echo "$(tasks.get-pict-file.results.pict-file)" > main.pict
+              cat main.pict
+              pict ./main.pict -o:1 -f:json > $(results.configs-json.path)
```
Comment (jsmid1) [inline integration-tests/pipelines/e2e-main-pipeline.yaml:66]: Could we make the order of combinations into a param?

Comment (rhopp) [reply integration-tests/pipelines/e2e-main-pipeline.yaml:66]: Definitely. I would just prefer to do that in followup tasks/PRs... (there is lot of room for all kinds of improvements still)

**Comment by rhopp** [comment]
The tests are expected to fail (they succeeded when I had the temp commits here - which are now removed).
I'll merge this PR now.

**Pull request merged**